allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -0,0 +1,305 @@
+from allensdk.api.cloud_cache.manifest import Manifest
+from allensdk.api.cloud_cache.file_attributes import CacheFileAttributes  # noqa: E501
+from allensdk.api.cloud_cache.utils import file_hash_from_path  # noqa: E501
+from allensdk.api.cloud_cache.utils import bucket_name_from_uri  # noqa: E501
+from allensdk.api.cloud_cache.utils import relative_path_from_uri  # noqa: E501
+
+
+class CloudCache(object):
+    """
+    A class to handle the downloading and accessing of data served from a cloud
+    storage system
+
+    Parameters
+    ----------
+    cache_dir: str or pathlib.Path
+        Path to the directory where data will be stored on the local system
+    """
+
+    _bucket_name = None
+
+    def __init__(self, cache_dir):
+        self._manifest = Manifest(cache_dir)
+        self._s3_client = None
+        self._manifest_file_names = self._list_all_manifests()
+
+    @property
+    def file_id_column(self) -> str:
+        """
+        The column in the metadata files used to uniquely
+        identify data files
+        """
+        return self._manifest.file_id_column
+
+    @property
+    def version(self) -> str:
+        """
+        The version of the dataset currently loaded
+        """
+        return self._manifest.version
+
+    @property
+    def metadata_flie_names(self) -> list:
+        """
+        List of metadata file names associated with this dataset
+        """
+        return self._manifest.metadata_file_names
+
+    @property
+    def s3_client(self):
+        if self._s3_client is None:
+            s3_config = Config(signature_version=UNSIGNED)
+            self._s3_client = boto3.client('s3',
+                                           config=s3_config)
+        return self._s3_client
+
+    @property
+    def manifest_file_names(self) -> list:
+        """
+        Sorted list of manifest file names associated with this
+        dataset
+        """
+        return copy.deepcopy(self._manifest_file_names)
+
+    def _list_all_manifests(self) -> list:
+        """
+        Return a list of all of the file names of the manifests associated
+        with this dataset
+        """
+        output = []
+        continuation_token = None
+        keep_going = True
+        while keep_going:
+            if continuation_token is not None:
+                subset = self.s3_client.list_objects_v2(Bucket=self._bucket_name,  # noqa: E501
+                                                        Prefix='manifests/',
+                                                        ContinuationToken=continuation_token)  # noqa: E501
+            else:
+                subset = self.s3_client.list_objects_v2(Bucket=self._bucket_name,  # noqa: E501
+                                                        Prefix='manifests/')
+
+            if 'Contents' in subset:
+                for obj in subset['Contents']:
+                    output.append(os.path.basename(obj['Key']))
```

Comment:
any reason not `pathlib.Path(obj['Key']).name` ?

Reply:
No reason. I'm only just learning `pathlib` and still feel much more comfortable in `os`. I'll make the changes.

Reply:
Done